addons/im_livechat/models/chatbot_script.py

```diff
@@ -0,0 +1,201 @@
+
+    @api.depends('script_step_ids.step_type')
+    def _compute_first_step_warning(self):
+        for script in self:
+            allowed_first_step_types = [
+                'question_selection',
+                'question_email',
+                'question_phone',
+                'free_input_single',
+                'free_input_multi',
+            ]
+            welcome_steps = script.script_step_ids and script._get_welcome_steps()
+            if welcome_steps and welcome_steps[-1].step_type == 'forward_operator':
+                script.first_step_warning = 'first_step_operator'
+            elif welcome_steps and welcome_steps[-1].step_type not in allowed_first_step_types:
+                script.first_step_warning = 'first_step_invalid'
+            else:
+                script.first_step_warning = False
+
+    @api.returns('self', lambda value: value.id)
+    def copy(self, default=None):
+        """ Correctly copy the 'triggering_answer_ids' field from the original script_step_ids to the clone.
+        This needs to be done in post-processing to make sure we get references to the newly created
+        answers from the copy instead of references to the answers of the original.
+
+        This implementation assumes that the order of created steps and answers will be kept between
+        the original and the clone, using 'zip()' to match the records between the two. """
+
+        default = default or {}
+        default['title'] = self.title + _(' (copy)')
+
+        clone_chatbot_script = super().copy(default=default)
+        if 'question_ids' in default:
+            return clone_chatbot_script
+
+        answers_map = {
+            original_answer: clone_answer
+            for clone_answer, original_answer
+            in zip(clone_chatbot_script.script_step_ids.answer_ids, self.script_step_ids.answer_ids)
+        }
+
+        for clone_step, original_step in zip(clone_chatbot_script.script_step_ids, self.script_step_ids):
+            clone_step.write({
+                'triggering_answer_ids': [
+                    (4, answer.id)
+                    for answer in [
+                        answers_map.get(original_answer)
+                        for original_answer
+                        in original_step.triggering_answer_ids
+                    ]
+                ]
+            })
+
+        return clone_chatbot_script
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        operator_partners_values = [self._prepare_operator_partner_values(
+            vals['title'],
+            vals.get('image_1920', False),
+        ) for vals in vals_list if 'operator_partner_id' not in vals and 'title' in vals]
+
+        operator_partners = self.env['res.partner'].create(operator_partners_values)
+
+        for vals, partner in zip(
+            [vals for vals in vals_list if 'operator_partner_id' not in vals and 'title' in vals],
```

Comment:
Maybe this could be done just once at the top, to make it clearer that `operator_partners_values` and this are in fact working on the same dataset?